include/aspect/parameters.h

```diff
@@ -293,6 +293,24 @@ namespace aspect
      */
     unsigned int                   n_compositional_fields;
     std::vector<std::string>       names_of_compositional_fields;
+
+    /**
+     * A vector that contains the field method for every compositional
+     * field. Consequently the vector has n_compositional_fields entries.
+     */
+    std::vector<std::string> compositional_field_methods;
```

Comment:
I think it would be better to make this a vector of enums, i.e., do the translation at parse time, rather than every time you have to look it up.